### DISCUSSION TITLE
React with TailwindCSS

### DISCUSSION THREAD

**Discussion by vicainelli** [cypress-io/cypress#17654] [Component Testing] [answered]
Does anyone know how can I load the TailwindCSS from the testing files?

I've tried to use the same approach I used on VueJS, importing the css file, but it does just not load the styles.

Here's the commit where I added the cypress component testing
https://github.com/vicainelli/cypress-component-testing-react-tailwindcss/commit/2fa25833cb965fadfeda6c53b80a23bb12b3b1c5

**Comment by mufasa71** [comment]
in mount there are options you can pass, one of theme is stylesheet, for example

```
mount(<App />, { stylesheet: "https://unpkg.com/tailwindcss@^2/dist/tailwind.min.css" });
```

**Reply by vicainelli** [reply]
Yeah, this works fine, thanks! 

But I would like to make it work with the local css file, because there are some custom properties.

**Reply by mufasa71** [reply]
For custom you can do the same but with slightly different approach:

Build your css:

```
npx tailwindcss -i ./src/tailwind.css -o ./dist/tailwind.css
```

Run server:

```
npx serve ./dist
```

And change mount:

```
mount(<App />, { stylesheet: "http://localhost:3000/tailwind.css" });
```
But I am pretty sure there could be better approach than this.

**Reply by JessicaSachs** [reply]
We do support importing css from the spec file. It should work exactly like it does in your source code.

**Comment by JessicaSachs** [comment, marked as answer]
Tailwind is getting loaded, but it's not being transpiled.

![Screen Shot 2021-08-18 at 6 02 01 PM](https://user-images.githubusercontent.com/2801156/129978736-24e5e9f4-9566-4ca0-a14a-2ad824dc5974.png)

You need to pass in your `craco` config. See our craco example here https://github.com/cypress-io/cypress/blob/bcefdaba6d6fc9705eeb5aef8b2711379bbb72b6/npm/react/examples/craco/cypress/plugins/index.js#L1-L9

**Reply by vicainelli** [reply]
Hey, Jessica,  thanks.

I did exacly this.

Firstly I tried a temporary fix by adding the compiled TailwindCSS style.

And now I'm using this one, by adding Craco on Cypress plugins file.

Btw, there's a typo error on [Cypress docs](https://docs.cypress.io/guides/references/migration-guide#Component-Testing).

Should be:
```js
require('tailwindcss/dist/tailwind.min.css')
```

Instead of:
```js
require('tailwindcss/dist/tailwindcss.min.css')
```

**Reply by JessicaSachs** [reply]
Thanks for the catch.

**Reply by JessicaSachs** [reply]
Fixed https://github.com/cypress-io/cypress-documentation/pull/4069
